src/lib/csvParser.js

```diff
@@ -0,0 +1,116 @@
+import parseCsv from "csv-parse/lib/sync";
+import camelize from "camelize";
+
+export const parseData = csv => {
+  const jsResults = camelize(
+    parseCsv(csv, {
+      columns: true,
+      skip_empty_lines: true
+    })
+  );
+
+  /*
+    { precinctData: { [precinctId]: dataForPrecinct }}
+  */
+  const precinctResults = jsResults.reduce((acc, current) => {
+    const precinctData = rowToData(current);
+    return { ...acc, [precinctData.key]: precinctData };
+  }, {});
+
+  return {
+    precinctResults
+  };
+};
+
+// Replace later if needed- looking for a standard way to
+const precinctIdentifier = rowObject => {
+  const { precinctFull } = rowObject;
+  return `${precinctFull}`;
+};
```

Comment:
ideally this would be the same as the geojson ids we've talked about. for now it's mostly indirection: in the new resulting `data.precinctResults` object the format will be `{ [precinctIdentifier(rowObject)]: rowToData(rowObject }`